File: docs-samples/data-engineering/GraphQL/C#/sample-graphql-app.cs
using System;
using Microsoft.Identity.Client;
using GraphQL.Client.Http;
using GraphQL.Client.Serializer.Newtonsoft;

namespace AzureADAuthApp
{
    class Program
    {
        private const string ClientId = "1e1126bf-17a1-4c9c-bcbe-8e41700991d6";  // This is your Application ID from Entra App Registration
        private const string TenantId = "cb4ed31a-dc29-496c-979a-ef47810cfeae";  // This is your Tenant ID from Entra App Registration
        private const string Authority = "https://login.microsoftonline.com/" + TenantId;
        private const string GraphQLUri = "https://api.fabric.microsoft.com/v1/workspaces/[WORKSPACE_ID]/graphqlapis/[API_ID]/graphql"; // This is your GraphQL API endpoint

        static async System.Threading.Tasks.Task Main(string[] args)
        {
            var app = PublicClientApplicationBuilder.Create(ClientId)
                .WithAuthority(Authority)
                .WithRedirectUri("http://localhost:1234")
                .Build();

            var scopes = new[] { "https://analysis.windows.net/powerbi/api/.default" };

            try
            {
                var result = await app.AcquireTokenInteractive(scopes).ExecuteAsync();
                if (result != null)
                {
                    Console.WriteLine("Authentication Success");

                    var graphQLClient = new GraphQLHttpClient(GraphQLUri, new NewtonsoftJsonSerializer());
                    
                    graphQLClient.HttpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", result.AccessToken);

                    var query = new GraphQLHttpRequest
                      {
                          Query = @"query {
                                        addresses {
                                            items {
                                                City
                                            }
                                        }
                                    }" 
                      };
                    
                    var graphQLResponse = await graphQLClient.SendQueryAsync<dynamic>(query);

                    Console.WriteLine(graphQLResponse.Data.ToString());
                }
            }
            catch (MsalServiceException ex)
            {
                Console.WriteLine($"Authentication Error: {ex.Message}");
            }
        }
    }
}

File: docs-samples/data-engineering/GraphQL/Python/sample-graphql-app.py
import asyncio
from msal import PublicClientApplication
from gql import Client, gql
from gql.transport.aiohttp import AIOHTTPTransport

# Constants
CLIENT_ID = "1e1126bf-17a1-4c9c-bcbe-8e41700991d6"  # Your Application ID
TENANT_ID = "cb4ed31a-dc29-496c-979a-ef47810cfeae"  # Your Tenant ID
AUTHORITY = f"https://login.microsoftonline.com/{TENANT_ID}"
GRAPHQL_URI = "https://api.fabric.microsoft.com/v1/workspaces/[WORKSPACE_ID]/graphqlapis/[API_ID]/graphql"  # GraphQL API Endpoint
SCOPES = ["https://analysis.windows.net/powerbi/api/.default"]  # Scopes for the resource you're accessing

async def main():
    # Setup MSAL Public Client
    app = PublicClientApplication(CLIENT_ID, authority=AUTHORITY)
    
    # Acquire token interactively
    result = None
    accounts = app.get_accounts()
    if accounts:
        result = app.acquire_token_silent(SCOPES, account=accounts[0])
    if not result:
        result = app.acquire_token_interactive(SCOPES)
    
    if "access_token" in result:
        print("Authentication Success")
        
        # Setup GraphQL client
        transport = AIOHTTPTransport(url=GRAPHQL_URI, headers={"Authorization": f"Bearer {result['access_token']}"})
        client = Client(transport=transport, fetch_schema_from_transport=True)
        
        # GraphQL query
        query = gql("""
            query {
                addresses {
                    items {
                        City
                    }
                }
            }
        """)
        
        # Execute query
        response = await client.execute_async(query)
        print(response)
    else:
        print(f"Authentication Error: {result.get('error_description')}")

if __name__ == "__main__":
    asyncio.run(main())